src/main/java/solrocr/OcrHighlighter.java

```diff
@@ -435,10 +437,22 @@ public OcrHighlightResult[] highlightOcrFields(
             // Stop highlighting
             break docLoop;
           } catch (RuntimeException e) {
-            // This catch-all prevents OCR highlighting from failing the complete query, instead
-            // users
-            // get an error message in their Solr log.
-            log.error("Could not highlight OCR content for document", e);
+            // This catch-all prevents OCR highlighting from failing the complete query,
+            // instead users get an error message in their Solr log.
+            if (content.getPointer() != null) {
+              List<SourcePointer.FileSource> fSource = content.getPointer().sources;
+              if (fSource != null && fSource.get(docId) != null) {
```

Comment:
I don't understand this? How does the internal Lucene docId map to the array of file sources parsed from the pointer? This is most certainly going to throw an `IndexOutOfBoundsException` in any real-world usage (where the docIds can be very large integers).

Reply:
I hoped the `docId` was somehow related to the files that get concatenated since it was in my case rather small (113 for a print with 200+ pages), but yes, that was just me misunderstanding.